IPv4 and IPv6 domain docs.

Enmk opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Other
Docs

Short description (up to few sentences):
IPv4 and IPv6 domains docs.
...

Enmk committed:
IPv4 and IPv6 domain docs.

blinkov commented on docs/en/data_types/domains/domains.md:
```diff
@@ -0,0 +1,24 @@
+# Domains
+
+Domains are special-purpose types, that add some extra features atop of existing 'base' type, leaving on-wire and on-disc format of underlying table intact. At the moment, Clickhouse does not support user-defined domains, and builtin domains are limited in functionality.
```
* Click**H**ouse
* I'd remove "and builtin domains are limited in functionality"
* Why single quotes around base?

blinkov commented on docs/en/data_types/domains/domains.md:
```diff
@@ -0,0 +1,24 @@
+
+Domains are special-purpose types, that add some extra features atop of existing 'base' type, leaving on-wire and on-disc format of underlying table intact. At the moment, Clickhouse does not support user-defined domains, and builtin domains are limited in functionality.
+
+You can use domains in anywhere corresponding base type can be used:
```
"in anywhere" sounds incorrect, maybe "in any place where"?

blinkov commented on docs/en/data_types/domains/domains.md:
```diff
@@ -0,0 +1,24 @@
+* Explicit column type name in `SHOW CREATE TABLE` or `DESCRIBE TABLE`
+* Custom input format for `INSERT INTO domain_table(domain_column) VALUES(...)`
+* Custom output format for `SELECT domain_column FROM domain_table`
+* Loading data from external source: `INSERT INTO domain_table FORMAT CSV ...`
```
Actually these are called input/output formats in the rest of documentation, not the features from previous two bullets. Those are more like "Input from human-readable representation for INSERT INTO..." and "Output to human-readable representation for SELECT...".

Enmk replied:
Fixed

blinkov commented on docs/en/data_types/domains/IPv4.md:
```diff
@@ -0,0 +1,179 @@
+## IPv4
+`IPv4` is a domain based on `UInt32` type, this is basically a typed replacement of `UInt32` for storing IPv4 values. It provides compact storage (compared to `String`), human-friendly output (compared to `UInt32`) and column type information on inspection.
+
+### Basic usage
```
Title case for all English headers: Basic **U**sage
Here and in many places below

Enmk replied:
Done

blinkov commented on docs/en/query_language/functions/ip_address_functions.md:
```diff
@@ -147,6 +147,70 @@ SELECT
 
+## toIPv4(string)
+
+An alias to `IPv4StringToNum()` that takes a string form of IPv4 address and returns value of [`IPv4`](../../data_types/domains/IPv4.md) type, which is binary equal to value returned by `IPv4StringToNum()`.
```
remove backticks around IPv4

Enmk replied:
Done

blinkov commented on docs/en/data_types/domains/domains.md:
```diff
@@ -0,0 +1,24 @@
+# Domains
```
* New files should be added into `toc_en.yml`
* Not-yet-translated languages should have similar files as relative symlinks to English documents and also added to `toc_*.yml`
* domains/domains.md will have weird URL with double "domains", so it'd be better to rename to overview.md or something like that

Enmk replied:
Fixed

blinkov commented:
Also there are new errors in links integrity check:

> INFO:root:Link to nowhere: #../../data_types/domains/IPv4/
> INFO:root:Link to nowhere: #../../data_types/domains/IPv6/

(probably because files are named in lowercase)

BayoNet commented on docs/en/data_types/domains/IPv4.md:
```diff
@@ -0,0 +1,66 @@
+## IPv4
+`IPv4` is a domain based on `UInt32` type, this is basically a typed replacement of `UInt32` for storing IPv4 values. It provides compact storage (compared to `String`), human-friendly output (compared to `UInt32`) and column type information on inspection.
```
IPv4 doesn't look as `SQL domain`. It looks like usual embedded data type, while domain is a custom entity, that user define within their database with `CREATE DOMAIN` clause.

Enmk replied:
Yep, I've mention that in the `overview.md`, `IPv4` and `IPv6` are kind of builtin domains, while there is still much work to do in order to support full-fledged user-defined domains.

blinkov commented on docs/en/data_types/domains/overview.md:
```diff
@@ -0,0 +1,26 @@
+* Call functions with values of domain column
+* etc.
+
+### Extra features of domains
```
Extra **F**eatures of **D**omains

Enmk replied:
Done

blinkov reviewed (changes requested):
+ it looks like some previous comments are not addressed yet

Enmk committed:
A script to set symlinks to docs missing in other languages.

Enmk commented on docs/en/data_types/domains/IPv4.md:
````diff
@@ -0,0 +1,179 @@
+```
+
+
+If you want to convert IPv4 value to a string, you have to do that explicitly with `IPv4NumToString()` function:
````
Fixed

Enmk commented on docs/en/data_types/domains/IPv4.md:
````diff
@@ -0,0 +1,179 @@
+```
+
+### Functions
+Any function that takes `UInt32` arguments can also take `IPv4` arguments too
````
Fixed that by removing the whole section.

Enmk commented on docs/en/data_types/domains/IPv4.md:
````diff
@@ -0,0 +1,179 @@
+└───────────────────────────┴────────────────┴─────────────────────────────┘
+```
+
+!!! warning BUG
````
Done

Enmk commented on docs/en/data_types/domains/IPv6.md:
```diff
@@ -0,0 +1,136 @@
+## IPv6
+`IPv6` is a domain based on `FixedString(16)` type, this is basically a typed replacement of `FixedString(16)` for storing IPv6 values. It provides compact storage (compared to `String`), human-friendly output (compared to `FixedString(16)`) and column type information on inspection.
```
Done

Enmk commented on docs/en/data_types/domains/IPv6.md:
```diff
@@ -0,0 +1,136 @@
+
+### Loading data from external source
+
+You can make Clikchouse load `IPv6` values from external source with `INSERT INTO ... FORMAT`:
```
Removed that section

Enmk commented:
* Simplified documentation to meet style of other docs - removed superfluous sections.
* Added doc files to toc_{ru,en,zh,fa}.yml
* Added docs in other languages as symbolic links to English docs. 
* Added a script that helps in adding links.
